cockroachcloud/cloud-sso-sql.md

```diff
@@ -0,0 +1,140 @@
+	~~~
+
+## Authenticate software users (service accounts) with external IdPs
+
+Currently, {{ site.data.products.db }} can only serve as a token issuer for human users. Authenticating service accounts, i.e. user identities to be controlled by software applications or scripts, rather by humans, is considerably more complicated, as it requires the user, or more realistically, an IdP admin, to provision the appropriate JWT token.
+
+{{ site.data.products.db }} SSO supports the use of external IdPs such as Google, Microsoft, GitHub, or customer deployed OIDC or SAML solutions, such as Okta. All of these options support multifactor authentication (MFA).
+
+**Learn more:**
+
+This [Cockroach Labs blog post](https://www.cockroachlabs.com/blog/) covers and provides further resources for a variety of auth token-issuing use cases, including using Okta and Google Cloud Platform to issue tokens.
+
+**Prerequisites:**
+
+- You must have the ability to create identities and issue tokens with a SAML or OIDC identity provider.
+	{{site.data.alerts.callout_success}}
+	The `issuer` and `audience` fields in the token must match values [configured in your cluster settings](#configure-your-cluster-to-accept-your-external-identity-provider).
+	{{site.data.alerts.end}}
+- You must have a user identity on a {{ site.data.products.db }} organization. For help setting up an organization and cluster, see: [Quickstart with CockroachDB](quickstart.html).
+- You must have access to a member of the [`admin` role](../{{site.versions["stable"]}}/security-reference/authorization.html#admin-role) in order to update cluster settings, which is required to add an external token issuer/IdP.
+- A SQL user specifically corresponding to the service account must be pre-provisioned on the cluster (or you must have access to a SQL role allowing you to create such a user).
+
+### Configure your cluster to accept your external identity provider
+
+In order to authenticate a service account to a {{ site.data.products.db }} cluster using a JWT issuer, you must update several cluster settings in the `server.jwt_authentication` namespace:
+
+1. `server.jwt_authentication.jwks`
+
+	Add your IdP's public signing key to your cluster's list of accepted signing JSON web keys (JWKS), under the `jwks` setting. This is a [JWK](https://www.rfc-editor.org/rfc/rfc7517) formatted single key or key set, containing the [public keys](../{{site.versions["stable"]}}/security-reference/transport-layer-security.html#key-pairs) for SSO token issuers/IdPs that will be accepted by your cluster.
+
+	By default, your cluster's configuration will contain the {{ site.data.products.db }}'s own public key, allowing {{ site.data.products.db }} to serve as an IdP. When modifying this cluster setting, you must include the {{ site.data.products.db }} public key in the key set. Failing to do so can prevent maintenance access by essential {{ site.data.products.db }} managed service accounts, leading to unintended consequences.
```

Comment:
```suggestion
	By default, your cluster's configuration will contain the {{ site.data.products.db }}'s own public key, allowing {{ site.data.products.db }} to serve as an IdP. When modifying this cluster setting, you must include the {{ site.data.products.db }} public key in the key set. Failing to do so can prevent maintenance access by the {{ site.data.products.db }} managed service, leading to unintended consequences. It can also break the cluster SSO for human users to this cluster.
```